gravitee-apim-console-webui/src/management/api/proxy/failover/api-proxy-failover.component.ts

```diff
@@ -0,0 +1,121 @@
+  styles: [require('./api-proxy-failover.component.scss')],
+})
+export class ApiProxyFailoverComponent implements OnInit, OnDestroy {
+  private unsubscribe$: Subject<boolean> = new Subject<boolean>();
+
+  private failoverForm: FormGroup;
+  public initialFailoverFormValue: ProxyFailover;
+
+  constructor(
+    private readonly formBuilder: FormBuilder,
+    @Inject(UIRouterStateParams) private readonly ajsStateParams,
+    private readonly apiService: ApiService,
+    private readonly snackBarService: SnackBarService,
+    private readonly permissionService: GioPermissionService,
+  ) {}
+
+  ngOnInit(): void {
+    this.apiService
+      .get(this.ajsStateParams.apiId)
+      .pipe(
+        takeUntil(this.unsubscribe$),
+        tap((api) => {
+          const isReadOnly =
+            !this.permissionService.hasAnyMatching(['api-definition-u']) || api.definition_context?.origin === 'kubernetes';
+          this.createForm(isReadOnly, api.proxy?.failover);
+          this.setupDisablingFields();
+        }),
+      )
+      .subscribe();
+  }
+
+  ngOnDestroy() {
+    this.unsubscribe$.next(true);
+    this.unsubscribe$.unsubscribe();
+  }
+
+  private createForm(isReadOnly: boolean, failover?: ProxyFailover) {
+    const isFailoverReadOnly = isReadOnly || !failover;
+
+    this.failoverForm = this.formBuilder.group({
+      enabled: [{ value: !!failover, disabled: isReadOnly }, []],
+      maxAttempts: [{ value: failover?.maxAttempts ?? null, disabled: isFailoverReadOnly }, [Validators.required]],
+      retryTimeout: [{ value: failover?.retryTimeout ?? null, disabled: isFailoverReadOnly }, [Validators.required]],
+    });
+    this.initialFailoverFormValue = this.failoverForm.getRawValue();
+  }
+
+  private setupDisablingFields() {
+    const controlKeys = ['maxAttempts', 'retryTimeout'];
+    this.enabled.valueChanges.pipe(takeUntil(this.unsubscribe$)).subscribe((checked) => {
+      controlKeys.forEach((k) => {
+        return checked ? this.failoverForm.get(k).enable() : this.failoverForm.get(k).disable();
+      });
+    });
+  }
+
+  onSubmit() {
+    const { enabled, maxAttempts, retryTimeout } = this.failoverForm.getRawValue();
+    const failover: ProxyFailover = enabled
+      ? {
+          maxAttempts,
+          retryTimeout,
+        }
+      : undefined;
+
+    return this.apiService
+      .get(this.ajsStateParams.apiId)
+      .pipe(
+        takeUntil(this.unsubscribe$),
+        switchMap(({ proxy, ...api }) =>
+          this.apiService.update({
+            ...api,
+            proxy: {
+              ...proxy,
+              failover,
+            },
+          }),
+        ),
+        tap(() => this.snackBarService.success('Configuration successfully saved!')),
+        catchError(({ error }) => {
+          this.snackBarService.error(error.message);
+          return EMPTY;
+        }),
+        tap(() => this.ngOnInit()),
+      )
+      .subscribe();
+  }
+
+  public get enabled() {
```

Comment:
I often place the getter on top of the controller. How about adding the `.value` here to avoid having it in the template. WDYT ?